src/sync/scheduler.ts

```diff
@@ -0,0 +1,144 @@
+import { Repository, Subscription } from "models/subscription";
+import { Task, TaskType } from "~/src/sync/sync.types";
+import { RepoSyncState } from "models/reposyncstate";
+import { getTargetTasks } from "~/src/sync/installation";
+import { createInstallationClient } from "utils/get-github-client-config";
+import Logger from "bunyan";
+import { booleanFlag, BooleanFlags } from "config/feature-flags";
+
+const getCursorKey = (type: TaskType) => `${type}Cursor`;
+const getStatusKey = (type: TaskType) => `${type}Status`;
+
+// These numbers were obtained experimentally by syncing a large customer with 60K quota from GitHub
+const RATE_LIMIT_QUOTA_PER_TASK_RESERVE = 500;
```

Comment:
/nit
Some thinkings, from ops' perspective, we want to adjust our apps somehow during runtime on the fly. So those "knobs", should we always use a FF to control it? Or should there be an admin api to control it? Should we use stash to control it (but re-deploy is required)? Or just hard code it in this case. 

We may want to talk about it for improve ops process. But for now I am okay with hardcoded.

Reply:
The idea is to have these small/aggressive enough so we would use all quota (not to slug behind GitHub dropping their rate-limiting counters).  Once we reach the ceiling, preemptive rate-limiting will make it chill.

But yeah, it could be adjust later, just don't think it will make too much difference. This would be very-very fine-grained tuning :)